src/main/java/org/assertj/core/api/AbstractBufferAssert.java

```diff
@@ -0,0 +1,175 @@
+   * // ... but this one fails as "buffer" is empty.
+   * Buffer buffer = ByteBuffer.wrap("".getBytes());
+   * assertThat(buffer).isNotEmpty();</code></pre>
+   *
+   * @return {@code this} assertion object.
+   * @throws AssertionError if the actual {@code Buffer} is {@code null}.
+   * @throws AssertionError if the actual {@code Buffer} has not been flipped.
+   * @throws AssertionError if the actual {@code Buffer} is empty.
+   */
+  public SELF isNotEmpty() {
+    isFlipped();
+    if (actual.limit() == 0) throwAssertionError(shouldNotBeEmpty());
+    return myself;
+  }
+
+  /**
+   * Verifies that the actual {@code Buffer} has a length equal to the given expected value.
+   *
+   * Example:
+   *
+   * <pre><code class='java'> byte[] testArray = "test".getBytes();
+   * Buffer buffer = ByteBuffer.wrap(testArray);
+   *
+   * // this assertion succeeds ...
+   * assertThat(buffer).hasLength(testArray.length);
+   *
+   * // ... but this one fails as "buffer" has a different length than the given expected value.
+   * assertThat(buffer).hasLength(testArray.length - 1);</code></pre>
+   *
+   * @param expected integer value representing the expected length of the buffer.
+   * @return {@code this} assertion object.
+   * @throws AssertionError if the actual {@code Buffer} is {@code null}.
+   * @throws AssertionError if the actual {@code Buffer} has not been flipped.
+   * @throws AssertionError if the actual {@code Buffer} length is different than the given expected value.
+   */
+  public SELF hasLength(int expected) {
+    isFlipped();
+    if (actual.limit() != expected) throwAssertionError(shouldHaveLength(expected, actual.limit(), actual));
+    return myself;
+  }
+
+  /**
+   * Verifies that the actual {@code Buffer} has a remaining length equal to the given expected value.
+   *
+   * The remaining length is the amount of space the buffer still has left, or in other words the space in the buffer
+   * that is not filled. It is defined as the capacity of the buffer minus the limit of the buffer.
+   *
+   * Example:
+   *
+   * <pre><code class='java'> byte[] testArray = "test".getBytes();
+   * ByteBuffer buffer = ByteBuffer.allocate(10);
+   * buffer.put(testArray);
+   * buffer.flip();
+   *
+   * // this assertion succeeds ...
+   * assertThat(buffer).hasRemainingLength(10 - testArray.length);
+   *
+   * // ... but this one fails as "buffer" has a different remaining length than the given expected value.
+   * assertThat(buffer).hasRemainingLength(10);</code></pre>
+   *
+   * @param expected integer value representing the expected length of the buffer.
+   * @return {@code this} assertion object.
+   * @throws AssertionError if the actual {@code Buffer} is {@code null}.
+   * @throws AssertionError if the actual {@code Buffer} has not been flipped.
+   * @throws AssertionError if the actual {@code Buffer}s remaining length is different than the given expected value.
+   */
+  public SELF hasRemainingLength(int expected) {
```

Comment:
Remaining at `Buffer` is defined as

> the number of elements between the current position and the limit.

This means that the verification is to check how many readable bytes are left.

If you want to verify remaining capacity (that's what this method is doing), then let's rename this method to `hasRemainingCapacity()`